pkg/operator/operator/k8s.go

```diff
@@ -385,6 +481,74 @@ func ONNXPredictorContainers(api *spec.API) ([]kcore.Container, []kcore.Volume)
 	return containers, volumes
 }
 
+func getTaskEnvVars(api *spec.API, container string) []kcore.EnvVar {
+	envVars := []kcore.EnvVar{
+		{
+			Name:  "CORTEX_KIND",
+			Value: api.Kind.String(),
+		},
+		{
+			Name:  "CORTEX_LOG_LEVEL",
+			Value: strings.ToUpper(api.TaskDefinition.LogLevel.String()),
+		},
+	}
+
+	for name, val := range api.TaskDefinition.Env {
+		envVars = append(envVars, kcore.EnvVar{
+			Name:  name,
+			Value: val,
+		})
+	}
+
+	if container == APIContainerName {
+		envVars = append(envVars,
+			kcore.EnvVar{
+				Name: "HOST_IP",
+				ValueFrom: &kcore.EnvVarSource{
+					FieldRef: &kcore.ObjectFieldSelector{
+						FieldPath: "status.hostIP",
+					},
+				},
+			},
+			kcore.EnvVar{
+				Name:  "CORTEX_PROJECT_DIR",
+				Value: path.Join(_emptyDirMountPath, "project"),
+			},
+			kcore.EnvVar{
+				Name:  "CORTEX_CACHE_DIR",
+				Value: _specCacheDir,
+			},
+			kcore.EnvVar{
+				Name:  "CORTEX_API_SPEC",
+				Value: config.BucketPath(api.Key),
+			},
+		)
+
+		cortexPythonPath := path.Join(_emptyDirMountPath, "project")
+		if api.TaskDefinition.PythonPath != nil {
+			cortexPythonPath = path.Join(_emptyDirMountPath, "project", *api.TaskDefinition.PythonPath)
+		}
+		envVars = append(envVars, kcore.EnvVar{
+			Name:  "CORTEX_PYTHON_PATH",
+			Value: cortexPythonPath,
+		})
+
+		if api.Compute.Inf > 0 {
```

Comment:
Made the following changes:
* Fixed the above `if` branch because the `api.Predictor` field was called that would have otherwise led to a panic.
* Removed nginx graceful stopper from the API's lifecycle. Wasn't really executing, so there was no worry anyway.